src/macaron/slsa_analyzer/checks/provenance_commit_check.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module adds a check that determines whether the repository URL came from provenance."""
+import logging
+
+from sqlalchemy import ForeignKey, String
+from sqlalchemy.orm import Mapped, mapped_column
+
+from macaron.database.table_definitions import CheckFacts
+from macaron.errors import ProvenanceError
+from macaron.repo_finder.provenance_extractor import extract_repo_and_commit_from_provenance
+from macaron.slsa_analyzer.analyze_context import AnalyzeContext
+from macaron.slsa_analyzer.checks.base_check import BaseCheck
+from macaron.slsa_analyzer.checks.check_result import CheckResultData, CheckResultType, Confidence, JustificationType
+from macaron.slsa_analyzer.registry import registry
+from macaron.slsa_analyzer.slsa_req import ReqName
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class ProvenanceDerivedCommitFacts(CheckFacts):
+    """The ORM mapping for justifications in the commit from provenance check."""
+
+    __tablename__ = "_provenance_derived_commit_check"
+
+    #: The primary key.
+    id: Mapped[int] = mapped_column(ForeignKey("_check_facts.id"), primary_key=True)  # noqa: A003
+
+    #: The reason this test failed to pass, when applicable.
+    error: Mapped[str] = mapped_column(String, nullable=True, info={"justification": JustificationType.TEXT})
```

Comment:
I believe you have switched to this `error` attribute to store the explanatory message from the check (as discussed in this conversation https://github.com/oracle/macaron/pull/704#discussion_r1593378468). Should we remove the `commit_digest` attribute as it's not being used anymore :thinking: ? This same comment applies for : 
https://github.com/oracle/macaron/blob/50b6e2b708b91518be965d60416c97acc0f996bd/src/macaron/slsa_analyzer/checks/provenance_repo_check.py#L22

Reply:
The `commit_digest` attribute is still being used in the `PASSED` case.